src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -800,7 +800,8 @@ public Task<HtmlString> RenderSectionAsync(string name, bool required)
             else if (required)
             {
                 // If the section is not found, and it is not optional, throw an error.
-                throw new InvalidOperationException(Resources.FormatSectionNotDefined(sectionName, Path));
+                var message = Resources.FormatSectionNotDefined(sectionName, ViewContext.View.Path);
```

Comment:
Use `ExecutingFilePath` which is updated as different pages execute. Will in this case mean the offending `_Layout.cshtml` file is identified, not the loaded view.